### ISSUE TITLE
JavaRuntimePropertiesDetector should not rely on nonstandard java CLI option -XshowSettings

### ISSUE THREAD

**Issue by acsipak** [karakun/OpenWebStart#464]
JavaRuntimePropertiesDetector relies on nonstandard java CLI option -XshowSettings. A fallback method to determine java.vendor, java.version, os.name and os.arch should be provided in case -XshowSettings is not implemented.

java.version can be parsed from the output of "java -version". Maybe os.arch, too - althought it does not look that straightforward.

(Context: I'm trying to make OWS work with old Oracle JRE versions such as 6 and 7. My company uses severely outdated software based on old Java versions. Turns out -XshowSettings was introduced in Java 8...)

**Comment by acsipak** [comment]
Update: I had a look into how OpenJDK 8 implements the -XshowSettings:properties argument of the java command. Hte following is somewhat theoretical, since I'm not an expert on JDK/JVM internals.

The java command is written in C, I found the source code here: https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/bin/java.c

The part relevant to the XshowSettings argument seems to be the ShowSettings method at around ln1470:

`/*
 * Prints all the Java settings, see the java implementation for more details.
 */
static void
ShowSettings(JNIEnv *env, char *optString)`

I'm not really into C, but it looks to me that this method calls the showSettings() static method of the sun.launcher.LauncherHelper Java class. Source: https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/launcher/LauncherHelper.java

This Java code parses the argument, and then produces the specified output. In our case, the last part of the solution is the printProperties() static method. The first line of the method is a System.getProperties(); - all the rest is just sorting and output handling.

So it looks to me the java command simply starts a Java class in a JVM and then outputs System.getProperties().

I'd guess a fallback solution for cases where -XshowSettings is not implemented could be starting a Java class that does the same. The tricky part is to do the task in a way that is compatible with most JVM implementations and platforms.

**Comment by sclassen** [comment]
OWS is compiled with Java 8. As a consequence the binaries we distribute will not run on a Java 7 JVM.
If you have an Oracle 7 or older than the WebStart of Oracle is available for these runtimes.
The simplest solution would be to use the WebStart bundled with these JVMs instead of trying to make OWS Java 6 compatible.

**Comment by acsipak** [comment]
Thing is, we did not switch to OWS from Oracle javaws because the latter became deprecated / unsupported in newer releases.

OWS has much more to offer in usability, UX, transparency (sane logging!), flexibility (handle multiple JVMs), open licensing (so I can understand error messages better by looking at the source). OWS is JWS done right, that's why we started using it.

I'd rather fork your code than go back to the crappy old Oracle javaws.
